fastapi_opa/auth/auth_saml.py

```diff
@@ -46,7 +107,7 @@ async def single_sign_on(auth: OneLogin_Saml2_Auth) -> RedirectResponse:
     @staticmethod
     async def assertion_consumer_service(
         auth: OneLogin_Saml2_Auth, request_args: Dict
-    ) -> Union[RedirectResponse, Dict]:
+    ) -> Union[RedirectResponse, Userdata]:
```

Comment:
This breaks the interface since this response will be returned to the OPA middleware.
I am in favor of using a pydantic model here but it will introduce a major change in the package (it also means changing the interface and the OIDC injectable) and is not really necessary for this change.

If you really want to use a pydantic model here you could deserialize it before returning with `return **Userdata`.

But you only changed the typehint anyways but still using a dictionary. So I'm not sure what it is all about yet.

Reply:
Changed back